lib/puppet_x/coi/jboss/provider/securitydomain/abstract_provider.rb

```diff
@@ -0,0 +1,47 @@
+# A module for JBoss security domain common abstract provider
+class Puppet_X::Coi::Jboss::Provider::SecurityDomain::AbstractProvider
+
+  def initialize provider
+    @provider = provider
+  end
+  COMMAND_SPLITTER = ','
+  NEWLINE_REPLACEMENT = ' '
+
+  # Creates a parametrized command to be executed by provider
+  # @return {String} a complete command without profile
+  def make_command_templates
+    resource = @provider
+    res = resource.instance_variable_get(:@resource)
+    correct_cmd = correct_command_template_begining(res)
+    options = []
+    res[:moduleoptions].keys.sort.each do |key|
+      value = res[:moduleoptions][key]
+      val = value
+      # FIXME: After coi-gov-pl/puppet-jboss#59 is resolved the fallowing lines
+      # should be moved to mungle function in securitydomain type not provider
+      val = 'undefined' if val.nil?
```

Comment:
I it a copy-paste from somewhere?! If so, please extract this to separate class (in the whole code base)